### DISCUSSION TITLE
Full Typescript module or namespace support

### DISCUSSION THREAD

**Discussion by dhoepelman** [scala-tsi/scala-tsi#190] [Ideas]
* [ ] Allow subpackages to be emitted as separate modules
* [ ] Emit in a namespaces/module/class?

**Comment by dhoepelman** [comment]
Abandoned attempt in
https://github.com/scala-tsi/scala-tsi/pull/49

**Comment by mariussoutier** [comment]
Are there still any efforts on this? I usually have structures like the following snippet in my projects and scala-tsi will just overwrite one with the other, so there's only one `IResponse` and no reference to any of surrounding objects.

```
object Api {
  class UserApi {
    case class Response()
  }
  class AdminApi {
    case class Response()
  }
}
```
